src/scaffoldmaker/utils/shieldmesh.py

```diff
@@ -606,3 +664,1178 @@ def generateElements(self, fieldmodule, coordinates, startElementIdentifier, mes
                         meshGroup.addElement(element)
 
         return elementIdentifier
+
+
+class ShieldMesh3D:
+    """
+        Generates a 3D shield mesh.
+    """
+
+    def __init__(self, elementsCountAcross, elementsCountRim, shieldMode=ShieldShape3D.SHIELD_SHAPE_OCTANT_PPP, box_derivatives=None):
+        """
+        3D shield structure can be used for a sphere mesh. 3 hex mesh merges to one box in the corner located in the centre.
+        The structure is a 3D version of the 2D shield structure. It has a 'quadruple point', a unique node on the
+        surface where the elements merge. 'Triple curves' meet at the quadruple point and a fourth curve that connects
+        to another quadruple point which is inside.
+         The structure is like a elementsCountAcross[0] X elementsCountAcross[1] X elementsCountAcross[2] box where
+        some nodes does not exist and stored as None. The quadruple point is stored at [n3z][0][n1z] (top, front and right most node)
+        where n3z is top most and n1z is the right most indexes.
+         Triple curves and surfaces connecting to the quadruple point divide the exterior surface and interior region into 3 regions
+        (top, left and right). Triple curve 1 connects the quadruple point to the plane 2-3. Similarly, triple curves 2 and 3
+        connect the quadruple point to the planes 1-3 and 1-2, respectively. It is the same for the inside quadruple.
+        Any point on region left has n2 = 0. Similarly on region right -> n1 = n1z and on top -> n3 = n3z.
+        There is a gap between node indexes of a node on a triple curve and the next nodes on the surface.
+
+        :param elementsCountAcross: number of elements as a list [elementsCountAcrossAxis1, elementsCountAcrossAxis2, elementsCountAcrossAxis3]
+        :param elementsCountRim:
+        :param shieldMode: It can be full or just a section of the whole shield.
+        """
```

Comment:
Missing param box_derivatives. I saw a similar description in sphere mesh but it doesn't properly explain how to use it. Can use 2-3 lines if needed.